typescript/test/unit/baseline.ts

```diff
@@ -79,10 +77,83 @@ describe('CodeGeneratorTest', () => {
           `-I${PROTOS_DIR} ` +
           ECHO_PROTO_FILE
       );
-      assert.strictEqual(
-        fs.readFileSync(GENERATED_CLIENT_FILE).toString(),
-        fs.readFileSync(CLIENT_LIBRARY_BASELINE).toString()
-      );
+      // put all baseline files into fileStack
+      let fileStack: string[] = [];
+      const dirStack: string[] = [];
+      putAllBaselineFiles(BASELINE_DIR, fileStack, dirStack);
+      // store every item (file or directory with full path in output dir and baseline dir) in stack, pop once a time
+      const protoItemStack: Item[] = [];
+      putItemToStack(protoItemStack, OUTPUT_DIR, BASELINE_DIR);
+      while (protoItemStack.length !== 0) {
+        const item = protoItemStack.pop();
+        if (!item) continue;
+        // if item is a file, compare it with baseline
+        if (fs.lstatSync(item.outputPath).isFile()) {
+          checkIdenticalFile(
+            item.outputPath,
+            item.baselinePath + BASELINE_EXTENSION
+          );
+          fileStack = fileStack.filter(file => {
+            file === (item.baselinePath + BASELINE_EXTENSION)
+          });
+        } else if(fs.lstatSync(item.outputPath).isDirectory()) {
+          // if item is a directory, loop the folder and put every item in stack again
+          putItemToStack(protoItemStack, item.outputPath, item.baselinePath);
+        }
+      }
+      assert.strictEqual(fileStack.length, 0);
     });
   });
 });
+
+function checkIdenticalFile(outputFullPath: string, baselineFullPath: string) {
+  assert(fs.existsSync(baselineFullPath))
+  assert.deepStrictEqual(
```

Comment:
This will generate a LOT of output. Also, it will fail on the first assert.
Maybe we can just save all the failures (filenames of non-existent files and filenames that don't match) to an array and then in the test just `assert(checkResultArray.length === 0)`.

Just thinking about the user: the output of `npm test` should be small enough to understand what's going on, and how many baseline files differ.

Reply:
Very helpful suggestions! Output error of not existing files and not identical files separately, which will not block the test.